packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -55,27 +60,41 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 		private readonly rebaser: Rebaser<TChange>,
 		private readonly changeFamily: ChangeFamily<TEditor, TChange>,
 		private readonly anchors: AnchorSet,
+		parentUndoRedoManager: UndoRedoManager<TChange, TEditor>,
+		repairDataStoreFactory: () => RepairDataStore,
```

Comment:
It's a bit annoying to have to pass in this stuff, both because it just adds boiler plate code, but also because it means the child branch gets access to the parent UndoRedoManager, which it could keep a copy of an do weird things to.

I think you can avoid it by making the constructor call self-referential:
```
		const fork: SharedTreeBranch<TEditor, TChange> = new SharedTreeBranch(
			() => this.head,
			(forked) => {...},
			this.sessionId,
			this.rebaser,
			this.changeFamily,
			anchors,
			this.undoRedoManager.clone(repairDataStoreFactory, (change) =>
				fork.applyChange(change, false),
			),
		);
```